src/test/integration/tests/basic.ts

```diff
@@ -12,232 +12,234 @@
-          assert.strictEqual((dom.querySelector('div') as any).bar, false);
-        }
+        args: [{}],
+        html: '<div>[object Object]</div>'
       }
     ],
+    // Objects are not dirty-checked before being toString()'ed
+    expectMutationsOnFirstRender: true,
     stableSelectors: ['div'],
   },
 
-  'two expressions in one property': {
-    render(x: any, y: any) {
-      return html`<div .foo="${x},${y}"></div>`;
+  'NodePart accepts an object with a toString method': {
+    render(x: any) {
+      return html`<div>${x}</div>`;
     },
     expectations: [
       {
-        args: [1, true],
-        html: '<div></div>',
-        check(assert: Chai.Assert, dom: HTMLElement) {
-          assert.strictEqual((dom.querySelector('div') as any).foo, '1,true');
-        }
+        args: [{toString() { return 'toString!'; }}],
+        html: '<div>toString!</div>'
       },
       {
-        args: [2, false],
-        html: '<div></div>',
-        check(assert: Chai.Assert, dom: HTMLElement) {
-          assert.strictEqual((dom.querySelector('div') as any).foo, '2,false');
-        }
+        args: [{toString() { return 'toString2!'; }}],
+        html: '<div>toString2!</div>'
       }
     ],
+    // Objects are not dirty-checked before being toString()'ed
+    expectMutationsOnFirstRender: true,
     stableSelectors: ['div'],
   },
 
-  'event binding': {
-    render(listener: (e: Event) => void) {
-      return html`<button @click=${listener}>X</button>`;
+  'NodePart accepts a function': {
+    render(x: any) {
+      return html`<div>${x}</div>`;
     },
     expectations: [
       {
-        args: [(e: Event) => (e.target as any).__wasClicked = true],
-        html: '<button>X</button>',
-        check(assert: Chai.Assert, dom: HTMLElement) {
-          const button = dom.querySelector('button')!;
-          button.click();
-          assert.strictEqual((button as any).__wasClicked, true);
-        }
+        args: [() => { throw new Error(); }],
+        html: '<div>() => { throw new Error(); }</div>'
       },
       {
-        args: [(e: Event) => (e.target as any).__wasClicked2 = true],
-        html: '<button>X</button>',
-        check(assert: Chai.Assert, dom: HTMLElement) {
-          const button = dom.querySelector('button')!;
-          button.click();
-          assert.strictEqual((button as any).__wasClicked2, true);
-        }
+        args: [() => { throw new Error("2"); }],
+        html: '<div>() => { throw new Error("2"); }</div>'
       }
     ],
-    stableSelectors: ['button'],
+    // Functions are not dirty-checked before being toString()'ed
+    expectMutationsOnFirstRender: true,
+    stableSelectors: ['div'],
   },
 
-  'boolean attribute binding, initially true': {
-    render(hide: boolean) {
-      return html`<div ?hidden=${hide}></div>`;
+  'NodePart accepts an element': {
+    // document.createElement is not shimmed in the server environment
```

Comment:
I'm not sure we even need this test. We should come up with a viable pattern for rendering nothing on the server and an element on the client though. I suspect we'll need some kind of `isServer` variable we can check.

Reply:
Yeah, I can remove this and/or turn it into an "expect to error during server-rendering" test.

Reply:
Removed for now.